### PR TITLE
Protected constructors accessible in subclass static methods

### LINKED ISSUE #9734: Protected constructors should be accessible on subclasses's static methods
I think the below should be allowed. 

```ts
class A {
    protected constructor() {

    }
}

class B extends A {
    static create() {
        new B(); // ERROR, but should not error.
    }
}
```
I wanted to override a static factory and customise it a bit. But it seems like I wasn't allowed to do it.

And my motivation is that private constructors are accessible on a class's static methods.
```ts
class A {
    static create() {
        new this();
    }
    protected constructor() {

    }
}
```
And a natural extension of the private case is to allow protected constructors on subclasses's static methods.

### PR DESCRIPTION
**Pull request by sandersn**
Fixes #9734

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -11544,8 +11544,23 @@ namespace ts {
...
             if (!isNodeWithinClass(node, declaringClassDeclaration)) {
+                const containingFunction = getContainingFunction(node);
+                const containingClass = getContainingClass(node);
+                if (containingClass) {
+                    const containingType = getTypeOfNode(containingClass);
+                    const baseTypes = getBaseTypes(<InterfaceType>containingType);
+                    if (baseTypes.length) {
+                        const baseType = baseTypes[0];
+                        if (containingFunction &&
+                            containingFunction.flags & NodeFlags.Static &&
```
Comment (RyanCavanaugh) [inline src/compiler/checker.ts]: Why does it matter whether the function is static?

Comment (sandersn) [inline src/compiler/checker.ts]: Oops. I wasn't paying attention and overfit to the example. I see that instance methods should be ok based on your constructor-with-super equivalence example.

Comment (sandersn) [inline src/compiler/checker.ts]: done

**Comment by RyanCavanaugh** [comment]
:+1:

**Pull request merged**